src/config_util.cc

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "config_util.h"
+#include "util.h"
+
+
+StatusOr<ConfigKV> ParseConfigLine(const std::string& line) {
+  enum { KEY, NORMAL, QUOTED, PRE_KEY_SPACE, AFTER_KEY_SPACE, AFTER_VAL_SPACE, ESCAPE, ERROR } state = PRE_KEY_SPACE;
```

Comment:
May we have a comment/document for this state machine?

Reply:
done